services/brig/src/Brig/Team/API.hs

```diff
@@ -230,69 +251,163 @@ createInvitation' tid mUid inviteeRole mbInviterUid fromEmail body = do
+        invitationId = storedInv.invitationId,
+        createdAt = storedInv.createdAt,
+        createdBy = storedInv.createdBy,
+        inviteeEmail = storedInv.email,
+        inviteeName = storedInv.name,
+        inviteeUrl = url
+      }
+
+mkInviteUrl ::
+  (Member TinyLog r) =>
+  ShowOrHideInvitationUrl ->
+  TeamId ->
+  InvitationCode ->
+  (Handler r) (Maybe (URIRef Absolute))
+mkInviteUrl HideInvitationUrl _ _ = pure Nothing
+mkInviteUrl ShowInvitationUrl team (InvitationCode c) = do
+  template <- invitationEmailUrl . invitationEmail . snd <$> teamTemplates Nothing
+  branding <- view App.templateBranding
+  let url = Text.toStrict $ renderTextWithBranding template replace branding
+  parseHttpsUrl url
+  where
+    replace "team" = idToText team
+    replace "code" = toText c
+    replace x = x
+    parseHttpsUrl :: (Member TinyLog r) => Text -> (Handler r) (Maybe (URIRef Absolute))
+    parseHttpsUrl url =
+      either (\e -> lift . liftSem $ logError url e >> pure Nothing) (pure . Just) $
+        parseURI laxURIParserOptions (encodeUtf8 url)
+    logError url e =
+      Log.err $
+        Log.msg @Text "Unable to create invitation url. Please check configuration."
+          . Log.field "url" url
+          . Log.field "error" (show e)
 
-getInvitation :: (Member GalleyAPIAccess r) => UserId -> TeamId -> InvitationId -> (Handler r) (Maybe Public.Invitation)
+getInvitation ::
+  ( Member GalleyAPIAccess r,
+    Member InvitationCodeStore r,
+    Member TinyLog r
+  ) =>
+  UserId ->
+  TeamId ->
+  InvitationId ->
+  (Handler r) (Maybe Public.Invitation)
 getInvitation uid tid iid = do
-  ensurePermissions uid tid [AddTeamMember]
-  showInvitationUrl <- lift $ liftSem $ GalleyAPIAccess.getExposeInvitationURLsToTeamAdmin tid
-  lift $ wrapClient $ DB.lookupInvitation showInvitationUrl tid iid
+  invitationM <- lift . liftSem $ Store.lookupInvitation tid iid
+  case invitationM of
+    Nothing -> pure Nothing
+    Just invitation -> do
+      ensurePermissions uid tid [AddTeamMember]
+      showInvitationUrl <- lift . liftSem $ GalleyAPIAccess.getExposeInvitationURLsToTeamAdmin tid
+      maybeUrl <- mkInviteUrl showInvitationUrl tid invitation.code
+      pure $ Just (Store.invitationFromStored maybeUrl invitation)
 
-getInvitationByCode :: Public.InvitationCode -> (Handler r) Public.Invitation
+getInvitationByCode ::
+  (Member Store.InvitationCodeStore r) =>
+  Public.InvitationCode ->
+  (Handler r) Public.Invitation
 getInvitationByCode c = do
-  inv <- lift . wrapClient $ DB.lookupInvitationByCode HideInvitationUrl c
-  maybe (throwStd $ errorToWai @'E.InvalidInvitationCode) pure inv
+  inv <- lift . liftSem $ Store.lookupInvitationByCode c
+  maybe (throwStd $ errorToWai @'E.InvalidInvitationCode) (pure . Store.invitationFromStored Nothing) inv
 
-headInvitationByEmail :: EmailAddress -> (Handler r) Public.HeadInvitationByEmailResult
-headInvitationByEmail e = do
+-- FIXME(mangoiv): This should not be in terms of store
```

Comment:
why not? what else? what does this function do?

Reply:
ah, cool. maybe next time you can be a little more explicit about the reasons, then you won't have to handle me separately :)